src/Promitor.Integrations.AzureMonitor/HttpPipelinePolicies/LogOutgoingRequestsPolicy.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using GuardNet;
+using Microsoft.Extensions.Logging;
+
+namespace Promitor.Integrations.AzureMonitor.HttpPipelinePolicies{
+    public class ModifyOutgoingAzureMonitorRequestsPolicy : HttpPipelinePolicy
+    {   
+        private readonly ILogger _logger;
+        public ModifyOutgoingAzureMonitorRequestsPolicy(ILogger logger)
+        {
+            Guard.NotNull(logger, nameof(logger));
+            _logger = logger;
+        }
+
+        public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {   
+            ModifyDateTimeParam(["starttime", "endtime"], message);
+            _logger.LogWarning("Modified URI: {uri}", message.Request.Uri.ToString());
+            await ProcessNextAsync(message, pipeline);
+        }
+
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            throw new NotSupportedException("Synchronous HTTP request path is not supported");
+        }
+
+        private void ModifyDateTimeParam(List<string> paramNames, HttpMessage message) 
+        {
+            // Modify the request URL by updating or adding a query parameter
+            var uriBuilder = new UriBuilder(message.Request.Uri.ToString());
+            var query = System.Web.HttpUtility.ParseQueryString(uriBuilder.Query);
+            foreach (var param in paramNames)
+            {
+                _logger.LogWarning("Original URI param {param} is {value}", param, query[param]);
```

Comment:
Will this be fore every request? If so, maybe use info given we're handling it?

Reply:
I changed the pipeline policy to outgoing request params, so this is no longer relevant. This policy will be used to modify batch requests.